pages/pipelines/environment_variables.md.erb

```diff
@@ -4,163 +4,448 @@ When the agent invokes your build scripts it passes in a set of standard Buildki
+
+Example: `"-v"`
+
+### BUILDKITE\_GIT_SUBMODULES
+
+The opposite of the value of the `no-git-submodules` [agent configuration option](https://buildkite.com/docs/agent/v3/configuration). The value cannot be modified as an environment variable.
+
+Values: `"true"` and `"false"`
+
+### BUILDKITE\_GITHUB\_DEPLOYMENT_ID
+
+The GitHub deployment ID. Only available on builds triggered by a [GitHub Deployment](https://developer.github.com/v3/repos/deployments/).
+
+Example: `"87972451"`
+
+### BUILDKITE\_GITHUB\_DEPLOYMENT_ENVIRONMENT
+
+The name of the GitHub deployment environment. Only available on builds triggered by a [GitHub Deployment](https://developer.github.com/v3/repos/deployments/).
+
+Example: `"production"`
+
+### BUILDKITE\_GITHUB\_DEPLOYMENT_TASK
+
+The name of the GitHub deployment task. Only available on builds triggered by a [GitHub Deployment](https://developer.github.com/v3/repos/deployments/).
+
+Example: `"deploy"`
+
+### BUILDKITE\_GITHUB\_DEPLOYMENT_PAYLOAD
+
+The GitHub deployment payload data as serialized JSON. Only available on builds triggered by a [GitHub Deployment](https://developer.github.com/v3/repos/deployments/).
+
+Example: `"production"`
+
+### BUILDKITE\_HOOKS_PATH
+
+The value of the `hooks-path` [agent configuration option](https://buildkite.com/docs/agent/v3/configuration). The value cannot be modified as an environment variable.
+
+Example: `"/etc/buildkite-agent/hooks/"`
+
+### BUILDKITE\_IGNORED_ENV
+
+A list of environment variables that have been set in your pipeline that are protected and will be overridden, used internally to pass data from the bootstrap to the agent. The value cannot be modified.
+
+Example: `"BUILDKITE\_GIT\_CLEAN_FLAGS"`
+
+### BUILDKITE\_JOB_ID
+
+The internal UUID Buildkite uses for this job. The value cannot be modified.
+
+Example: `"e44f9784-e20e-4b93-a21d-f41fd5869db9"`
+
+### BUILDKITE_LABEL
+
+The label/name of the current job. The value cannot be modified.
+
+Example: `"\:hammer\: Specs"`
+
+### BUILDKITE\_LAST\_HOOK\_EXIT_STATUS
+
+The exit code of the last hook that ran, used internally by the hooks. The value cannot be modified.
+
+Example: `"-1"`
+
+### BUILDKITE\_LOCAL\_HOOKS_ENABLED
+
+The opposite of the value of the `no-local-hooks` [agent configuration option](https://buildkite.com/docs/agent/v3/configuration). The value cannot be modified as an environment variable.
+
+Values: `"true"` and `"false"`
+
+### BUILDKITE_MESSAGE
+
+The message associated with the build, usually the commit message. The value cannot be modified.
+
+Example: `"Added a great new feature"`
+
+### BUILDKITE\_NO\_LOCAL_HOOKS
+
+Whether local hooks in the repository are allowed. This variable is read by the `buildkite-agent start` command ???. It can be set in the [agent config](https://buildkite.com/docs/agent/v3/configuration) file using the `no-local-hooks` option, or in by exporting this environment variable in the `environment`, `pre-checkout` or `pre-command` hooks.
```

Comment:
Are we keeping NO_LOCAL_HOOKS? Or is that one of the ones that doesn't show up anymore?